makeflow/src/dag_node.c

```diff
@@ -220,6 +249,209 @@ void dag_node_add_target_file(struct dag_node *n, const char *filename, char *re
 	target->created_by = n;
 }
 
+void dag_node_prepare_node_size(struct dag_node *n)
```

Comment:
I suggest to add documentation to these functions. In particular, why do we compute the size of the file differently whether it should exists or not?